### PR TITLE
homebrew: Move repeated logic from homebrew modules into module_utils

### LINKED ISSUE #8323: homebrew: Move repeated logic from homebrew modules into module_utils
### Summary

There is some validation logic across the homebrew modules that has been copy-pasted. This should be moved to module_utils to promote code re-use and maintainability.

I'm just making this issue as a formality. I'm already working on this now.

### Issue Type

Bug Report

### Component Name

homebrew

### Ansible Version

```console (paste below)
$ ansible --version
ansible [core 2.16.5]
  config file = None
  configured module search path = ['/Users/kit/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /opt/homebrew/Cellar/ansible/9.4.0_1/libexec/lib/python3.12/site-packages/ansible
  ansible collection location = /Users/kit/.ansible/collections:/usr/share/ansible/collections
  executable location = /opt/homebrew/bin/ansible
  python version = 3.12.2 (main, Feb  6 2024, 20:19:44) [Clang 15.0.0 (clang-1500.1.0.2.5)] (/opt/homebrew/Cellar/ansible/9.4.0_1/libexec/bin/python)
  jinja version = 3.1.3
  libyaml = True
```

### Community.general Version

```console (paste below)
$ ansible-galaxy collection list community.general

```

### Configuration

```console (paste below)
$ ansible-config dump --only-changed
# /Users/kit/.ansible/collections/ansible_collections
Collection        Version
----------------- -------
community.general 9.0.0

# /opt/homebrew/Cellar/ansible/9.4.0_1/libexec/lib/python3.12/site-packages/ansible_collections
Collection        Version
----------------- -------
community.general 8.5.0
```

(I'm developing on the 9.0.0 version.)

### OS / Environment

MacOS

### Steps to Reproduce

NA

### Expected Results

NA

### Actual Results

NA

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by kitizz**
Fixes #8323.

##### SUMMARY

There is some validation logic across the homebrew modules that has been copy-pasted. This should be moved to module_utils to promote code re-use and maintainability.

Fixes #8323.

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

homebrew

### CODE REVIEW

File: changelogs/fragments/8323-refactor-homebrew-logic-module-utils.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - "homebrew: Refactor common argument validation logic into dedicated ``module_utils/homebrew.py`` file."
```
Comment (felixfontein) [inline changelogs/fragments/8323-refactor-homebrew-logic-module-utils.yml:2]:
```suggestion
  - "homebrew, homebrew_cask - refactor common argument validation logic into a dedicated ``homebrew`` module utils (https://github.com/ansible-collections/community.general/issues/8323, https://github.com/ansible-collections/community.general/pull/8324)."
```

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution!

**Comment by felixfontein** [comment]
If nobody objects, I'll merge this on the weekend.

**Comment by kitizz** [comment]
Thanks for the review!

**Comment by felixfontein** [comment]
@kitizz thanks for your contribution!

**Pull request merged**